frontend/src/test/scala/bloop/bsp/BspProtocolSpec.scala

```diff
@@ -95,6 +96,42 @@ class BspProtocolSpec(
     }
   }
 
+  test("check the correct contents of jvm test environment") {
+    TestUtil.withinWorkspace { workspace =>
+      object Sources {
+        val `A.scala` =
+          """/A.scala
+            |object A
+          """.stripMargin
+      }
+
+      val logger = new RecordingLogger(ansiCodesSupported = false)
+      val jvmOptions = List("-DSOME_OPTION=X")
+      val jvmConfig = Some(Config.JvmConfig(None, jvmOptions))
+      val `A` = TestProject(
+        workspace,
+        "a",
+        List(Sources.`A.scala`),
+        jvmConfig = jvmConfig
+      )
+
+      val projects = List(`A`)
+      loadBspState(workspace, projects, logger) { state =>
+        val (stateA, result) = state.jvmTestEnvironment(`A`, None)
+        assert(stateA.status == ExitStatus.Ok)
+        val environmentItem = result.items.head
+        assert(result.items.size == 1)
+        assert(environmentItem.environmentVariables.contains("BLOOP_OWNER"))
```

Comment:
We generally prefer to use `assertNoDiff` to check invariants when possible. This check and the jvm options check are good candidates for `assertNoDiff`, could you change it to use that?